pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -1629,14 +1629,20 @@ void Object_beginTypeContext(JSOBJ _obj, JSONTypeContext *tc) {
     if (PyLong_Check(obj)) {
         PRINTMARK();
         tc->type = JT_LONG;
-        GET_TC(tc)->longValue = PyLong_AsLongLong(obj);
-
+        int _overflow = 0;
+        GET_TC(tc)->longValue = PyLong_AsLongLongAndOverflow(obj, &_overflow);
         exc = PyErr_Occurred();
```

Comment:
```suggestion
        exc = GET_TC(tc)->LongValue == -1 && PyErr_Occurred();
```

Then can simplify condition on next line to just if `(_overflow && !exc)`